projects/gloo/cli/pkg/cmd/install/installer_test.go

```diff
@@ -0,0 +1,325 @@
+package install_test
+
+import (
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	install2 "github.com/solo-io/gloo/pkg/cliutil/install"
+	"github.com/solo-io/gloo/projects/gloo/cli/pkg/cmd/install"
+	"github.com/solo-io/gloo/projects/gloo/cli/pkg/cmd/options"
+	"github.com/solo-io/gloo/projects/gloo/cli/pkg/constants"
+)
+
+type MockInstallClient struct {
+	expectedCrds     []string
+	applied          bool
+	waited           bool
+	resources        []install2.ResourceType
+	knativeInstalled bool
+	knativeOurs      bool
+}
+
+func (i *MockInstallClient) KubectlApply(manifest []byte) error {
+	Expect(i.applied).To(BeFalse())
+	i.applied = true
+	resources, err := install2.GetResources(string(manifest))
+	Expect(err).NotTo(HaveOccurred())
+	i.resources = resources
+	return nil
+}
+
+func (i *MockInstallClient) WaitForCrdsToBeRegistered(crds []string, timeout, interval time.Duration) error {
+	Expect(i.waited).To(BeFalse())
+	i.waited = true
+	Expect(crds).To(ConsistOf(i.expectedCrds))
+	return nil
+}
+
+func (i *MockInstallClient) CheckKnativeInstallation() (bool, bool, error) {
+	return i.knativeInstalled, i.knativeOurs, nil
+}
+
+var _ = Describe("Install", func() {
+
+	var (
+		file      string
+		installer install.GlooStagedInstaller
+		opts      options.Options
+		validator MockInstallClient
+	)
+
+	BeforeEach(func() {
+		file = filepath.Join(RootDir, "_test/gloo-test-unit-testing.tgz")
+		opts.Install.Namespace = "gloo-system"
+		opts.Install.HelmChartOverride = file
+	})
+
+	expectKinds := func(resources []install2.ResourceType, kinds []string) {
+		for _, resource := range resources {
+			Expect(kinds).To(ContainElement(resource.Kind))
```

Comment:
```suggestion
			ExpectWithOffset(1, kinds).To(ContainElement(resource.Kind))
```
To make ginkgo log the correct line numbers in case of failed expectations in helper functions (see [here](https://onsi.github.io/gomega/#making-assertions-in-helper-functions)).